kernel/src/process.rs

```diff
@@ -1263,7 +1264,7 @@ impl<C: Chip> ProcessType for Process<'_, C> {
             return None;
```

Comment:
Hmm...I think we should change the return type of `switch_to`. You're right that the arch crate can't differentiate between a timeslice expiration and any other interrupt, and therefore we shouldn't include the systick expired option in `ContextSwitchReason`. But the kernel still needs to know, so we need this to return something like `Option<ProcessStoppedReason>`.

Reply:
After some thought about my changes in #1767, it probably makes more sense for `do_process()` to check for timeslice expirations, so that if a scheduler called `do_process(no_systick)`, checking for timeslice expirations could be avoided. I updated the code so that `ContextSwitchReason` no longer has `TimesliceExpiration` as a variant, and instead check for expiration in the `Interrupts` match arm.